src/librustc_metadata/schema.rs

```diff
@@ -200,7 +200,7 @@ pub struct TraitImpls {
 #[derive(RustcEncodable, RustcDecodable)]
 pub struct Entry<'tcx> {
     pub kind: EntryKind<'tcx>,
-    pub visibility: ty::Visibility,
+    pub visibility: Lazy<ty::Visibility>,
```

Comment:
This makes sense, I only had it like that because it was always just an enum discriminant IIRC.